typescript/frontend-cicd/lib/frontend-cicd-stack.ts

```diff
@@ -0,0 +1,121 @@
+import { aws_cloudfront_origins, Duration, RemovalPolicy, Stack, StackProps } from 'aws-cdk-lib';
+import { CodePipeline, CodePipelineSource, ShellStep } from 'aws-cdk-lib/pipelines';
+import * as ecr from 'aws-cdk-lib/aws-ecr';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
+import * as codecommit from 'aws-cdk-lib/aws-codecommit';
+import * as codebuild from 'aws-cdk-lib/aws-codebuild';
+import * as codepipeline from 'aws-cdk-lib/aws-codepipeline';
+import * as codepipelineActions from 'aws-cdk-lib/aws-codepipeline-actions';
+import { Construct } from 'constructs';
+
+export class FrontendCicdStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    // The code that defines your stack goes here
+    const pipeLine = new codepipeline.Pipeline(this, 'MyFrontEndPipeline', {
+      pipelineName: 'MyFrontEndPipeline'
+    });
+
+    // Source
+    const repository = new codecommit.Repository(this, 'MyFrontEndRepo', {
+      repositoryName: 'MyFrontEndRepo',
+      description: 'My Frontend Repository'
+    });
+
+    const sourceOutput = new codepipeline.Artifact();
+    const sourceAction = new codepipelineActions.CodeCommitSourceAction({
+      actionName: 'CodeCommit',
+      repository: repository,
+      branch: 'master',   //or main
+      output: sourceOutput
+    });
+    pipeLine.addStage({
+      stageName: 'Source',
+      actions: [sourceAction],
+    });
+    //
+
+    // Build
+    const project = new codebuild.Project(this, 'MyFrontEndBuild', {
+      source: codebuild.Source.codeCommit({ repository }),
+      buildSpec: codebuild.BuildSpec.fromSourceFilename('./buildspec.yaml'),
+      environment: {
+        buildImage: codebuild.LinuxBuildImage.STANDARD_5_0,
+        privileged: true
+      },
+      environmentVariables: {
+        tag: {
+          value: 'cdk'
+        }
+      }
+    });
+
+    const buildOutput = new codepipeline.Artifact();
+    const buildAction = new codepipelineActions.CodeBuildAction({
+      actionName: 'CodeBuild',
+      project: project,
+      input: sourceOutput,
+      outputs: [buildOutput],
+      // executeBatchBuild: true,
+      // combineBatchBuildArtifacts: true
+    });
+    pipeLine.addStage({
+      stageName: 'Build',
+      actions: [buildAction],
+    });
+
+    // Change bucket name global unique.
+    const bucket = new s3.Bucket(this, '<YOUR UNIQUE BUCKET NAME>', {
```

Comment:
The CDK will disambiguate your bucket name with a random hash at the end if memory serves, but you should definitely tell the user about this in the README.

Reply:
As per your request, I have changed it to use hash.